powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/configuration/conditions/IsNotMssqlCondition.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * PowerAuth Server and related software components
+ * Copyright (C) 2024 Wultra s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package io.getlime.security.powerauth.app.server.configuration.conditions;
+
+import org.springframework.boot.jdbc.DatabaseDriver;
+import org.springframework.context.annotation.Condition;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.core.env.Environment;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+import org.springframework.lang.NonNull;
+
+import java.util.Objects;
+
+/**
+ * A condition that the datasource is not MSSQL.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+public class IsNotMssqlCondition implements Condition {
+
+    @Override
+    public boolean matches(ConditionContext context, @NonNull AnnotatedTypeMetadata metadata) {
+        final Environment environment = Objects.requireNonNull(context.getBeanFactory()).getBean(Environment.class);
+        final String url = environment.getProperty("spring.datasource.url");
+        return DatabaseDriver.fromJdbcUrl(url) != DatabaseDriver.SQLSERVER;
+    }
```

Comment:
🤔 We duplicate the whole logic with only a different comparison. There should be a way how to negate the condition, something like this.
```suggestion
public class IsNotMssqlCondition extends NoneOfNestedConditions {

    @IsMssqlCondition()
    static class IsMssql {
    }
```

Reply:
Sure, we can do this.